test/spec/Feature/Query/AggregateFunctionsSpec.hs

```diff
@@ -141,155 +141,364 @@ allowed =
+                {"supervisor": 1, "process_costs_count": 2},
+                {"supervisor": 2, "process_costs_count": 2},
+                {"supervisor": 3, "process_costs_count": 3},
+                {"supervisor": 4, "process_costs_count": 1}]|]
+              { matchHeaders = [matchContentTypeJson] }
+          it "works on nested resources grouped by spreaded fields" $ do
+            get "/process_supervisor?select=...processes(factory_id,...process_costs(count()))&order=processes(factory_id)" `shouldRespondWith`
+              [json|[
+                {"factory_id": 1, "count": 2},
+                {"factory_id": 2, "count": 4},
+                {"factory_id": 3, "count": 2}]|]
+              { matchHeaders = [matchContentTypeJson] }
+            get "/process_supervisor?select=...processes(factory:factory_id,...process_costs(process_costs_count:count()))&order=processes(factory)" `shouldRespondWith`
+              [json|[
+                {"factory": 1, "process_costs_count": 2},
+                {"factory": 2, "process_costs_count": 4},
+                {"factory": 3, "process_costs_count": 2}]|]
+              { matchHeaders = [matchContentTypeJson] }
+          it "works on different levels of the nested resources at the same time" $
+            get "/process_supervisor?select=...processes(factory:factory_id,processes_count:count(),...process_costs(process_costs_count:count()))&order=processes(factory)" `shouldRespondWith`
+              [json|[
+                {"factory": 1, "processes_count": 2, "process_costs_count": 2},
+                {"factory": 2, "processes_count": 4, "process_costs_count": 4},
+                {"factory": 3, "processes_count": 2, "process_costs_count": 2}]|]
+              { matchHeaders = [matchContentTypeJson] }
 
-      context "supports count() aggregate without specifying a field" $ do
-        it "works by itself in the embedded resource" $ do
-          get "/process_supervisor?select=supervisor_id,...processes(count())&order=supervisor_id" `shouldRespondWith`
+      context "to-many spread relationships" $ do
+        it "supports the use of aggregates on spreaded fields" $ do
+          get "/budget_categories?select=budget_owner,total_budget:budget_amount.sum(),...budget_expenses(total_expenses:expense_amount.sum())&order=budget_owner" `shouldRespondWith`
             [json|[
-              {"supervisor_id": 1, "count": 2},
-              {"supervisor_id": 2, "count": 2},
-              {"supervisor_id": 3, "count": 3},
-              {"supervisor_id": 4, "count": 1}]|]
+              {"budget_owner":"Brian Smith","total_budget":2000.42,"total_expenses":600.52},
+              {"budget_owner":"Dana de Groot","total_budget":500.33,"total_expenses":null},
+              {"budget_owner":"Jane Clarkson","total_budget":9000.53,"total_expenses":100.22},
+              {"budget_owner":"Sally Hughes","total_budget":500.23,"total_expenses":900.27}]|]
+            { matchHeaders = [matchContentTypeJson] }
```

Comment:
Doing an aggregate over an array is :fire: 

```
curl 'localhost:3000/budget_categories?select=budget_owner,...budget_expenses(total_expenses:expense_amount.sum())'

[{"budget_owner":"Brian Smith","total_expenses":600.52}, 
 {"budget_owner":"Sally Hughes","total_expenses":900.27}, 
 {"budget_owner":"Jane Clarkson","total_expenses":100.22}, 
 {"budget_owner":"Dana de Groot","total_expenses":null}]
```

Could we document this possibility on https://postgrest--3640.org.readthedocs.build/en/3640/references/api/aggregate_functions.html#aggregate-functions-with-columns-from-a-spreaded-resource ?

Reply:
Good idea, I'll add it there.

Reply:
Ok cool, then this should be good to merge now.